server/app/routes/_index.tsx

```diff
@@ -30,26 +49,148 @@ export async function action({ request }: ActionArgs) {
+      id: true,
+    },
+    where: {
+      name: {
+        search: searchTerm,
+        mode: "insensitive",
+      },
+    },
+    orderBy: {
+      _relevance: {
+        fields: ["name"],
+        search: searchTerm,
+        sort: "desc",
+      },
+    },
+    take: suggestionNumber,
+  });
+
+  return data.map((product) => ({
+    name: product.name,
+    imageUrl: getProductImageUrl(product.id),
+    id: product.id,
+  }));
+};
+
+function createSearchTerm(productName: string) {
+  return (
+    productName
+      .toLowerCase()
+      .trim()
+      .split(" ")
+      // :* used to search prefixes
+      .map((s) => `${s}:*`)
+      .join(" & ")
+  );
+}
+
 export default function Index() {
   const [searchProdName, setSearchProdName] = useState("");
+  const searchSuggestionFetcher = useFetcher<typeof loader>();
+  const [currentSuggestions, setCurrentSuggestions] = useState<Suggestion[]>(
+    []
+  );
+  const navigate = useNavigate();
+  const formRef = useRef<HTMLFormElement>(null);
 
   useEffect(() => {
-    setSearchProdName("");
-  }, []);
+    if (
+      searchSuggestionFetcher.type === "done" &&
+      searchSuggestionFetcher.data
+    ) {
+      setCurrentSuggestions(searchSuggestionFetcher.data);
+      console.log(searchSuggestionFetcher.data);
+    }
+  }, [searchSuggestionFetcher]);
 
   return (
     <div className="mx-4 h-full content-center items-center space-y-4 pt-[20vh] text-center md:container md:mx-auto">
       <h1 className="w-full text-5xl font-bold">Welcome to 6 Months Later!</h1>
       <p>Start by searching for a product:</p>
       <div className="mx-auto space-x-3 md:flex md:w-1/2">
-        <Form method="post" className="w-full">
-          <TextInput
-            className="mb-3"
-            placeholder="Search for a product..."
-            value={searchProdName}
-            onChange={(event) => setSearchProdName(event.target.value)}
-            name="productName"
-          />
+        <Form method="post" className="w-full" ref={formRef}>
+          <Combobox
+            as="div"
+            className="text-tremor-default relative w-full min-w-[10rem]"
+          >
+            <Combobox.Button className="tremor-TextInput-root relative mb-3 flex w-full w-full min-w-[10rem] items-center rounded-md border border-gray-300 bg-white text-gray-700 shadow-sm hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-blue-200">
+              <Combobox.Input
+                className="tremor-TextInput-input w-full border-0 bg-transparent py-2 pl-4 pr-4 text-sm font-medium placeholder:text-gray-500 focus:outline-none focus:ring-0"
+                placeholder="Search for a product..."
+                name="productName"
+                value={searchProdName}
+                onKeyDown={(event) => {
+                  if (event.key === "Enter") {
+                    event.preventDefault();
+                    event.stopPropagation();
+
+                    formRef.current?.submit();
```

Comment:
Would it make sense to use Remix's `useSubmit()` hook here?

https://remix.run/docs/en/main/hooks/use-submit

Reply:
Seems to work